lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -345,23 +356,51 @@ def run
       unless config[:disable_customization]
         # Wait for customization to complete
         puts 'Waiting for customization to complete...'
-        CustomizationHelper.wait_for_sysprep(vm, vim, Integer(get_config(:sysprep_timeout)), 10)
+        CustomizationHelper.wait_for_sysprep(@vm, vim, Integer(get_config(:sysprep_timeout)), 10)
         puts 'Customization Complete'
-        sleep 2 until vm.guest.ipAddress
-        connect_host = config[:fqdn] = config[:fqdn] ? get_config(:fqdn) : vm.guest.ipAddress
+        sleep 2 until connect_host
+        connect_host
       end
-      wait_for_access(connect_host, connect_port, protocol)
+      wait_for_access(connect_port, protocol)
```

Comment:
I think that removing the host argument from this method signature and having the method call `connect_host` itself might be giving this method too much responsibility. When I look at it, I see "connect to a host, I'll give you a port and a protocol", but I'm left wondering what the host is. When you look at the code, it calls `connect_host`. Ideally we'd look at a line like `wait_for_access(connect_host, connect_port, protocol)` and not need to look inside.